File: app/src/main/java/com/kinwatt/powermeter/common/AppSettings.java
package com.kinwatt.powermeter.common;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class AppSettings {
    private static final String KEY_QUESTIONARY_COMPLETED = "20180507_form";

    private static AppSettings instance;
    private SharedPreferences preferences;

    private Context context;

    private AppSettings(Context context) {
        this.context = context;
        preferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public static AppSettings getAppSettings(Context context) {
        if (instance == null) {
            instance = new AppSettings(context);
        }
        return instance;
    }

    public boolean isQuestionaryCompleted() {
        return preferences.getBoolean(KEY_QUESTIONARY_COMPLETED, false);
    }
    public void setQuestionaryCompleted(boolean completed) {
        preferences.edit().putBoolean(KEY_QUESTIONARY_COMPLETED, completed).apply();
    }
}

File: app/src/main/java/com/kinwatt/powermeter/common/AppSettings.kt
package com.kinwatt.powermeter.common

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

class AppSettings private constructor(private val context: Context) {
    private val preferences: SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)

    var isQuestionaryCompleted: Boolean
        get() = preferences.getBoolean(KEY_QUESTIONARY_COMPLETED, false)
        set(completed) = preferences.edit().putBoolean(KEY_QUESTIONARY_COMPLETED, completed).apply()

    companion object {
        private const val KEY_QUESTIONARY_COMPLETED = "20180507_form"

        private var instance: AppSettings? = null

        fun getAppSettings(context: Context): AppSettings {
            if (instance == null) {
                instance = AppSettings(context)
            }
            return instance!!
        }
    }
}
